packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -942,24 +945,27 @@ class RenderFlex extends RenderBox with ContainerRenderObjectMixin<RenderBox, Fl
     return defaultHitTestChildren(result, position: position);
   }
 
-  static const Color _black = const Color(0xBF000000);
-  static const Color _yellow = const Color(0xBFFFFF00);
-  static const double _kMarkerSize = 0.1;
-  static const TextStyle _debugMarkerTextStyle = const TextStyle(
-    color: const Color(0xFF900000),
-    fontSize: 7.5,
-    fontWeight: FontWeight.w800,
-  );
-  static Paint _debugMarkerPaint;
-  TextPainter _debugMarkerLabel;
-
-  bool _debugReportOverflow = true;
+  @override
+  String get overflowHints {
+    return 'The overflowing $runtimeType has an orientation of $_direction.\n'
+      'The edge of the $runtimeType that is overflowing has been marked '
+      'in the rendering with a yellow and black striped pattern. This is '
+      'usually caused by the contents being too big for the $runtimeType. '
+      'Consider applying a flex factor (e.g. using an Expanded widget) to '
+      'force the children of the $runtimeType to fit within the available '
+      'space instead of being sized to their natural size.\n'
+      'This is considered an error condition because it indicates that there '
+      'is content that cannot be seen. If the content is legitimately bigger '
+      'than the available space, consider clipping it with a ClipRect widget '
+      'before putting it in the flex, or using a scrollable container rather '
+      'than a Flex, for example using ListView.';
+  }
```

Comment:
I recommend passing this is an argument to the overflowPaintIndicator method instead of having a public getter that does this. It's a weird thing to expose on the RenderFlex API.

Reply:
Done.